### PR TITLE
Ci ops migration

### PR DESCRIPTION
**Pull request by johnalotoski**
Migration from old iohk-ops ci to new ci-ops repo and deployment.

### CODE REVIEW

File: modules/hydra-testing.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    ./modules/nix_nsswitch.nix
+    <nixpkgs/nixos/modules/virtualisation/amazon-image.nix>
+  ];
+  boot.loader.grub.device = lib.mkForce "/dev/nvme0n1";
+  _module.args.name = "hydra";
+  services.sshd.enable = true;
+  boot.loader.grub.splashImage = null;
+  services.postgresql.superUser = "root";
+  services.grafana.enable = true;
+  nix.binaryCaches = lib.mkForce [ "https://cache.nixos.org" "https://cache.nixos.org" "https://cache.nixos.org" ];
```
Comment (cleverca22) [inline modules/hydra-testing.nix]: dont need the duplicates here

File: modules/macs/guest/apply.sh [hunk 2]
```diff
@@ -85,14 +85,14 @@ EOF
     ls -la /private/var/run || true
     ln -s /private/var/run /run || true
     nix-channel --add https://nixos.org/channels/nixos-19.03 nixpkgs
-    nix-channel --add https://github.com/LnL7/nix-darwin/archive/master.tar.gz darwin
+    nix-channel --add https://github.com/LnL7/nix-darwin/archive/1d09aa2681bb3fbebb6cf9a40c21c6c871117146.tar.gz darwin # TODO, fix this
```
Comment (cleverca22) [inline modules/macs/guest/apply.sh]: `substituteAll` is ran over this file, so you can do `@rev@` and then nix can replace it with a value from the niv data

File: deployments/deployment-testing.nix [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  hydra = { resources, ... }: {
+    deployment.targetEnv = "ec2";
+    deployment.ec2.region = "eu-west-2";
+    deployment.ec2.instanceType = "m5.4xlarge";
+    deployment.ec2.keyPair = resources.ec2KeyPairs.todo;
```
[the remark this answers is not in the document]
Comment (johnalotoski) [reply deployments/deployment-testing.nix]: File deleted.

File: jobsets/cardano.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+let
+  iohkpkgs = import ../. { inherit pkgs; };
+  jobs = mapTestOn ((packagePlatforms iohkpkgs) // {
+    iohk-ops = [ "x86_64-linux" ];
+    iohk-ops-integration-test = [ "x86_64-linux" ];
+    nixops = [ "x86_64-linux" ];
+  });
+  cardanoSrc = pkgs.fetchgit cardano-sl-src;
+  cardano-sl-src = builtins.fromJSON (builtins.readFile ./../cardano-sl-src.json);
```
[the remark this answers is not in the document]
Comment (johnalotoski) [reply jobsets/cardano.nix]: cardanoSrc, cardano-sl-src, cardanoRelease attrs deleted.

File: modules/hydra-testing.nix (hunk 1 above)
Comment (johnalotoski) [reply modules/hydra-testing.nix]: Trimmed.

File: modules/macs/guest/apply.sh (hunk 2 above)
Comment (johnalotoski) [reply modules/macs/guest/apply.sh]: Substituted at the url level since the full url (account, repo, rev) may change with the switching of a fork.

**Pull request merged**